### PR TITLE
feat(metrics): MRI support for metrics API

### PR DESCRIPTION
**Pull request by obostjancic**
Part of: https://github.com/getsentry/getsentry/issues/10656

Enables usage of MRI as an identifier when querying the metrics API. 

Notable changes:
- _OrganizationMetricsEndpoint_ (metrics/meta) is redesigned and returns all available metrics.
- _OrganizationMetricsDataEndpoint_ (metrics/data) now supports passing MRIs in `field` query param. For example `field=sum(c:sessions/all@none)` is valid and equivalent to `field=sum(session.all)`

### CODE REVIEW

File: src/sentry/snuba/metrics/query_builder.py [hunk 1]
```diff
@@ -84,14 +85,36 @@
 from sentry.utils.snuba import parse_snuba_datetime
 
 
-def parse_field(field: str) -> MetricField:
+def parse_field(field: str, allow_mri: bool = False) -> MetricField:
```
Comment (iambriccardo) [inline src/sentry/snuba/metrics/query_builder.py:88]: Why are you not using `allow_mri`? Maybe you missed it

File: src/sentry/snuba/metrics/query_builder.py [hunk 2]
```diff
@@ -84,14 +85,36 @@
...
+    mri_matches = MRI_SCHEMA_REGEX.match(field) or MRI_EXPRESSION_REGEX.match(field)
```
Comment (iambriccardo) [inline src/sentry/snuba/metrics/query_builder.py:89]:
```suggestion
    mri_matches = allow_mri and (MRI_SCHEMA_REGEX.match(field) or MRI_EXPRESSION_REGEX.match(field))
```

File: src/sentry/snuba/metrics/query_builder.py [hunk 3]
```diff
@@ -84,14 +85,36 @@
...
+
+    if mri_matches:
+        return parse_mri_field(field)
+
+    return parse_public_field(field)
+
+
+def parse_mri_field(field: str) -> MetricField:
+
```
Comment (iambriccardo) [inline src/sentry/snuba/metrics/query_builder.py:98]:
```suggestion
```

**Review by iambriccardo — COMMENTED** [review]
Great work, left some minor comments!

File: src/sentry/snuba/metrics/query_builder.py (hunk 1 above)
Comment (obostjancic) [reply src/sentry/snuba/metrics/query_builder.py:88]: Yep, good catch!

File: src/sentry/snuba/metrics/query_builder.py (hunk 2 above)
Comment (obostjancic) [reply src/sentry/snuba/metrics/query_builder.py]: [c595fca](https://github.com/getsentry/sentry/pull/50404/commits/c595fca247cc7fa9ff561549da77ac3232b145ff)

**Review by iambriccardo — APPROVED** [review]
🚢

**Pull request merged**